open-api/rest-catalog-open-api.yaml

```diff
@@ -2800,6 +3057,53 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - options
+      properties:
+        select:
+          description: A list of the selected columns
+          type: array
+          items:
+            type: string
+        project:
+          description: A list of the projections
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        options:
+          type: object
+          additionalProperties:
+            type: string
+        snapshot-id:
+          type: integer
+          format: int64
+        from-snapshot-id:
+          type: integer
+          format: int64
+        to-snapshot-id:
+          type: integer
+          format: int64
+        timestamp-ms:
+          type: integer
+          format: int64
+        case-sensitive:
+          type: boolean
+
+    PlanTableRequest:
+      type: object
+      properties:
+        allOf:
+          $ref: '#/components/schemas/PreplanTableRequest'
+        plan-tasks:
```

Comment:
I think this should just be a single plan task, based on the devlist discussion

Reply:
Based on the devlist discussion it looks like a list? https://lists.apache.org/thread/flmw1qts0hv8n0k4pd9n1nfry322633y 

```
`Jack Ye - Wednesday, January 31, 2024 10:15:43 AM PST`
Sounds good. I don't really have any strong opinions here. So looks like we
are landing on this?


*PreplanTable: POST /v1/namespaces/ns/tables/t/preplan*{ "filter": {
"type": "in", "term": "x", "values": [1, 2, 3] }, "select": ["x", "a.b"] }

{ "plan-tasks": [ { ... }, { ... } ] } // opaque object

```
I can change this to just be an object instead of a list of objects if that is preferred?

Reply:
I am talking about this:

```
PlanTable w/ a plan task: POST /v1/namespaces/ns/tables/t/plan
{ "filter": {"type": "in", "term": "x", "values": [1, 2, 3] }, "select": ["x", "a.b"], "plan-task": { ... } }

{ "file-scan-tasks": [ { ... }, { ... } ] }
```

So a single plan task is in the request.

Reply:
Will fix this.